tests/test_views.py

```diff
@@ -0,0 +1,367 @@
+"""
+Module tests user-facing views of the Enterprise app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import os
+import unittest
+
+
+from django.core.urlresolvers import reverse, NoReverseMatch
+from django.shortcuts import render_to_response
+from django.test import Client
+import mock
+from pytest import mark, raises
+
+from enterprise.models import (
+    EnterpriseCustomerUser,
+    UserDataSharingConsentAudit,
+)
+from enterprise.utils import NotConnectedToEdX
+from enterprise.views import (
+    GrantDataSharingPermissions
+)
+from test_utils.factories import EnterpriseCustomerFactory, UserFactory
+
+
+def fake_render(template, context, request):  # pylint: disable=unused-argument
+    """
+    Switch the request to use the Django rendering engine instead of Mako.
+    """
+    return render_to_response('enterprise/grant_data_sharing_permissions.html', context=context)
+
+
+@mark.django_db
+class TestGrantDataSharingPermissions(unittest.TestCase):
+    """
+    Test GrantDataSharingPermissions.
+    """
+
+    url = reverse('grant_data_sharing_permissions')
+
+    @mock.patch('enterprise.views.add_lookup')
+    def test_patch_mako_lookup(self, add_lookup_mock):
+        """
+        Check that we attempt to patch mako with the correct path.
+        """
+        expected_path = os.path.abspath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)
+                ),
+                os.pardir,
+                'enterprise',
+                'templates',
+                'enterprise'
+            )
+        )
+        GrantDataSharingPermissions.patch_mako_lookup()
+        assert add_lookup_mock.call_args == (('main', expected_path,), {})
```

Comment:
Not sure if this test is valuable as written. A better test might be to simply check if the new template file exists on the filesystem in the absolute path that was passed to `add_lookup_mock`